core/src/main/java/org/eclipse/hono/util/TenantResult.java

```diff
@@ -0,0 +1,64 @@
+/**
+ * Copyright (c) 2018 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 1.0 which is available at
+ * https://www.eclipse.org/legal/epl-v10.html
+ *
+ * SPDX-License-Identifier: EPL-1.0
+ */
+
+package org.eclipse.hono.util;
+
+import io.vertx.core.json.JsonObject;
+import io.vertx.core.json.DecodeException;
+
+/**
+ * A container for the result returned by Hono's Tenant API.
+ *
+ */
+public final class TenantResult extends RequestResponseResult<JsonObject> {
```

Comment:
is this class any different from e.g. RegistrationResult?

Reply:
No, it is not - there are several things that need to be refactored, since code duplication would otherwise increase.
I would keep that outside of this PR, though (and no concerns, I am not forgetting these refactorings!).

IMHO we should have something like `RequestResponseResultJson extends RequestResponseResult<JsonObject>`, which then is used by the Tenant and the Registration API.
Would affect quite some places and should thus be contained in a different commit.
WDYT?

Reply:
yes, but we should name it `JsonRequestResponseResult` :-)

Reply:
:+1: